src/helperFunctions/process.py

```diff
@@ -88,23 +108,43 @@ def start_single_worker(process_index, label: str, function: Callable) -> Except
 
 def check_worker_exceptions(process_list: List[ExceptionSafeProcess], worker_label: str,
                             config: Optional[ConfigParser] = None, worker_function: Optional[Callable] = None) -> bool:
+    '''
+    Iterate over the `process_list` and check if exceptions occurred. In case of an exception, the process and its
+    children will be terminated. If ``throw_exceptions`` in the FACT configuration is set to `false`, the worker
+    may be restarted by passing a function (if the value is not set, the worker will not be restarted). In this case,
+    the function will always return ``False``. If ``throw_exceptions`` is set to `true` and an exception occurs,
+    the worker will not be restarted and the return value is ``True``.
+
+    :param process_list: A list of worker processes.
+    :param worker_label: A label used for logging (e.g. `Analysis` or `Unpacking`).
+    :param config: An instance of the FACT configuration.
+    :param worker_function: A function used for restarting the worker (optional).
+    :return: ``True`` if an exception occurred in any process and ``throw_exceptions`` in the FACT configuration is
+             set to `true` and ``False`` otherwise.
+    '''
     return_value = False
     for worker_process in process_list:
         if worker_process.exception:
             logging.error(color_string('Exception in {} process'.format(worker_label), TerminalColors.FAIL))
-            logging.error(worker_process.exception[1])
-            terminate_process_and_childs(worker_process)
+            _, stack_trace = worker_process.exception
```

Comment:
Not new code, I know, but would it not make sense to include the stack trace directly in the first error. Then it's also highlighted red, what I would say makes sense in this case.